### DISCUSSION TITLE
Transforming classic publishing news with custom page layout to modern news

### DISCUSSION THREAD

**Discussion by mohammadamer** [pnp/pnpframework#169] [Q&A]
I'm Transforming classic publishing news with custom page layout to modern news. With default mapping pages are transformed but empty without any content. How would it be in terms of mapping if the Classic old news are using Article Content Type and the modern news are using Site pages.

I'm wondering how should I do the mapping. Should I have the same old Page layout? should I have the same columns in the site pages content Type?

This is a snippet of xml mapping file that is related to the old page layout

` <PageLayout Name="XyzImageCenter" AssociatedContentType="" PageLayoutTemplate="AutoDetect" PageHeader="Custom">
      <Header Type="FullWidthImage" Alignment="Left" ShowPublishedDate="false">
        <Field Name="PublishingRollupImage" HeaderProperty="ImageServerRelativeUrl" Functions="ToImageUrl({PublishingRollupImage})" />
        <Field Name="ArticleByLine" HeaderProperty="TopicHeader" Functions="" />
        <Field Name="PublishingContact" HeaderProperty="Authors" Functions="ToAuthors({PublishingContact})" />
      </Header>
      <MetaData>
        <Field Name="Title" TargetFieldName="Title" Functions="" />
        <Field Name="Comments" TargetFieldName="Comments" Functions="" />
        <Field Name="PublishingStartDate" TargetFieldName="PublishingStartDate" Functions="" />
        <Field Name="PublishingExpirationDate" TargetFieldName="PublishingExpirationDate" Functions="" />
        <Field Name="PublishingContactEmail" TargetFieldName="PublishingContactEmail" Functions="" />
        <Field Name="PublishingContactName" TargetFieldName="PublishingContactName" Functions="" />
        <Field Name="PublishingContactPicture" TargetFieldName="PublishingContactPicture" Functions="" />
        <Field Name="PublishingIsFurlPage" TargetFieldName="PublishingIsFurlPage" Functions="" />
        <Field Name="ArticleStartDate" TargetFieldName="ArticleStartDate" Functions="" />
        <Field Name="Region" TargetFieldName="Region" Functions="" />
        <Field Name="Avdeling" TargetFieldName="Avdeling" Functions="" />
        <Field Name="Dokumenttype" TargetFieldName="Dokumenttype" Functions="" />
        <Field Name="Klassifisering" TargetFieldName="Klassifisering" Functions="" />
        <Field Name="Kunde" TargetFieldName="Kunde" Functions="" />
        <Field Name="Tagg" TargetFieldName="Tagg" Functions="" />
        <Field Name="Highlight" TargetFieldName="Highlight" Functions="" />
        <Field Name="ArticleEndDate" TargetFieldName="ArticleEndDate" Functions="" />
        <Field Name="Ingress" TargetFieldName="Ingress" Functions="" />
      </MetaData>
      <WebPartZones>
        <WebPartZone ZoneIndex="0" ZoneId="Footer" Row="1" Column="1" />
        <WebPartZone ZoneIndex="0" ZoneId="Right" Row="1" Column="1" />
        <WebPartZone ZoneIndex="0" ZoneId="xFooterZone0000000000000000000000" Row="1" Column="1" />
      </WebPartZones>
    </PageLayout>`

**Comment by jansenbe** [comment]
Please check out the options to configure your pagelayout mapping file as explained here: https://docs.microsoft.com/en-us/sharepoint/dev/transform/modernize-userinterface-site-pages-model-publishing. Also find a sample (the one I use for testing) mapping here: https://github.com/pnp/pnpframework/blob/dev/src/lib/PnP.Framework.Modernization.Test/Transform/Publishing/custompagelayoutmapping.xml
